client/user/requests.go

```diff
@@ -0,0 +1,38 @@
+package user
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/http"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/url"
+)
+
+type updateRequestBody struct {
+	UserRoles []string `json:"roles"`
+}
+
+func NewCreateRequest(ctx context.Context, client http.Client, createInp CreateUserInput) *http.Request {
+	url := url.CreateUser(client.BaseUrl(), createInp.Username)
+	body := fmt.Sprintf("roles=%s&isApiOnlyUser=%t", createInp.UserRoles, createInp.ApiOnlyUser)
```

Comment:
same as above, let's use the more general http header support

Reply:
Let's get that in in your PR if that' sok